### PR TITLE
Update the gradle task to add fallback scheme and host if needed when retrieving deep links.

### PR DESCRIPTION
**Pull request by hannah-hyj**
This is to support feature: https://github.com/flutter/devtools/issues/7541 

before behavior: do not show this link in the dev tool if there's no scheme or host
target behavior: show the link in the dev tool with a error text : ( missing scheme/ missing domain)

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Data Driven Fixes]: https://github.com/flutter/flutter/wiki/Data-driven-Fixes

### CODE REVIEW

File: packages/flutter_tools/gradle/src/main/groovy/flutter.groovy [hunk 1]
```diff
@@ -570,11 +570,12 @@ class FlutterPlugin implements Plugin<Project> {
                                         }
                                     }
                                 }
-                                schemes.each { scheme ->
-                                    hosts.each { host ->
-                                        if (!paths) {
-                                            appLinkSettings.deeplinks.add(new Deeplink(scheme: scheme, host: host, path: ".*", intentFilterCheck: intentFilterCheck))
-                                        } else {
+                                if(!hosts.isEmpty() || !paths.isEmpty()){
```
Comment (chunhtai) [inline packages/flutter_tools/gradle/src/main/groovy/flutter.groovy:573]: do we not fallback if the customers only have schemes?

Comment (hangyujin) [reply packages/flutter_tools/gradle/src/main/groovy/flutter.groovy:573]: yes,  the dev tool data table is listed by "domain view" "path view" "single url view" , if there is only schemes, we don't show it.

Comment (chunhtai) [reply packages/flutter_tools/gradle/src/main/groovy/flutter.groovy:573]: I think for our current design we can't surface this case, but I think this is something we should go over again with stakeholders.

This is not something we need to figure in this PR though.

**Review by chunhtai — APPROVED** [review]
LGTM

**Pull request merged**